pkg/agent/utils.go

```diff
@@ -263,6 +263,16 @@ func IsKubernetesVersionGe(actualVersion, version string) bool {
 	return v1.GE(v2)
 }
 
+// IsAzureLinuxCgroupv2 returns true if the VHD distro is Azure Linux and the Kubernetes version is greater than or equal to 1.27.0
+func IsAzureLinuxCgroupv2(config *datamodel.NodeBootstrappingConfiguration) bool {
+	if config.AgentPoolProfile.Distro.IsCBLMarinerDistro() || isMariner(config.OSSKU) {
```

Comment:
No need to check distro since isMariner(config.OSSKU) **should** already be true for every Mariner/AzureLinux distro.

Key word being should, this PR https://github.com/Azure/AgentBaker/pull/3399/files needs to go in first then you can drop the IsCBLMarinerDistro check.